scopesim/utils.py

```diff
@@ -919,3 +935,20 @@ def interp2(x_new, x_orig, y_orig):
         y_new = np.interp(x_new, x_orig[::-1], y_orig[::-1])
 
     return y_new
+
+
+def write_report(text, filename=None, output=["rst"]):
+    """ Writes a report string to file in latex or rst format"""
+    if isinstance(output, str):
+        output = [output]
+
+    if filename is not None:
```

Comment:
Because filename is passed (generally) from the Effects.report function. There if filename is None, the report RST text is returned by the function. If filename is not None, then the RST text is **also** written to file. filename=None by default in the Effects method